src/libsystemd/sd-event/sd-event.c

```diff
@@ -3354,6 +3354,10 @@ static int event_source_enter_ratelimited(sd_event_source *s) {
         if (s->ratelimited)
                 return 0; /* Already ratelimited, this is a NOP hence */
 
+        if (!exceeded) {
+                s->rate_limit.begin = now(CLOCK_MONOTONIC);
+        }
```

Comment:
coding style, no {} arund single line if block

also, why not just do this in  sd_event_source_enter_ratelimit(), i.e. in the caller?